unit12/unit12.js

```diff
@@ -0,0 +1,77 @@
+// Exercise 1 
+const compareArray = (array1 = [], array2 = []) => JSON.stringify(array2) === JSON.stringify(array1)
+
+// Exercise 2 
+const takeFlatArray = (array = []) => {
+  const flatArray = array.flat(Infinity)
+  return flatArray
+}
```

Comment:
3. Không được viết các lệnh dạng:
```javascript
  const condition = true
  if (condition) {
    return true
  } else {
    return false
  }
```

return thẳng luôn cái `condition`

4.
Vd:
```javascript
  const arr1 = [1, 2]
  const arr2 = [2, 1]
  
  // sau khi stringify
  const str1 = "[1, 2]"
  const str2 = "[2, 1]"
```
So sánh vẫn phải trả về `arr1` is equal `arr2`

Nếu stringify như bên trên sẽ bị sai